utils/test_results.py

```diff
@@ -76,132 +261,143 @@ def generate_test_results(
+
+        filtered_test_ids = set([test["test_id"] for test in totals])
+
+        test_ids = test_ids & filtered_test_ids if test_ids else filtered_test_ids
 
-            totals = totals.filter(test_id__in=test_ids)
-        case GENERATE_TEST_RESULT_PARAM.SKIPPED:
-            test_ids = (
-                totals.values("test")
-                .annotate(
-                    skip_count_sum=Sum("skip_count"),
-                    fail_count_sum=Sum("fail_count"),
-                    pass_count_sum=Sum("pass_count"),
+    if parameter is not None:
+        totals = DailyTestRollup.objects.filter(repoid=repoid, date__gt=since)
+
+        match parameter:
+            case GENERATE_TEST_RESULT_PARAM.FLAKY:
+                flakes = Flake.objects.filter(
+                    Q(repository_id=repoid)
+                    & (Q(end_date__date__isnull=True) | Q(end_date__date__gt=since))
                 )
-                .filter(skip_count_sum__gt=0, fail_count_sum=0, pass_count_sum=0)
-                .values("test_id")
-            )
 
-            totals = totals.filter(test_id__in=test_ids)
-        case GENERATE_TEST_RESULT_PARAM.SLOWEST:
-            num_tests = totals.distinct("test_id").count()
+                flaky_test_ids = set([flake.test_id for flake in flakes])
+
+                test_ids = test_ids & flaky_test_ids if test_ids else flaky_test_ids
+            case GENERATE_TEST_RESULT_PARAM.FAILED:
+                totals = DailyTestRollup.objects.filter(repoid=repoid, date__gt=since)
 
-            slowest_test_ids = (
-                totals.values("test")
-                .annotate(
-                    runtime=Avg("avg_duration_seconds")
-                    * (Sum("pass_count") + Sum("fail_count"))
+                failed_test_ids = (
+                    totals.values("test")
+                    .annotate(fail_count_sum=Sum("fail_count"))
+                    .filter(fail_count_sum__gt=0)
+                    .values("test_id")
                 )
-                .order_by("-runtime")
-                .values("test_id")[0 : slow_test_threshold(num_tests)]
-            )
+                failed_test_id_set = {test["test_id"] for test in failed_test_ids}
 
-            totals = totals.filter(test_id__in=slowest_test_ids)
+                test_ids = (
+                    test_ids & failed_test_id_set if test_ids else failed_test_id_set
+                )
+            case GENERATE_TEST_RESULT_PARAM.SKIPPED:
+                totals = DailyTestRollup.objects.filter(repoid=repoid, date__gt=since)
+
+                skipped_test_ids = (
+                    totals.values("test")
+                    .annotate(
+                        skip_count_sum=Sum("skip_count"),
+                        fail_count_sum=Sum("fail_count"),
+                        pass_count_sum=Sum("pass_count"),
+                    )
+                    .filter(skip_count_sum__gt=0, fail_count_sum=0, pass_count_sum=0)
```

Comment:
what are your thoughts on not having the fail_count_sum check here --- signaling tests which were constantly failing that were then skipped

Reply:
I guess it makes sense to have tests that have never passed here, but there's no way of knowing if it's like interleaved in the way that its failing and skipped, i'm cool with trying it though